src/main/java/network/brightspots/rcv/ContestConfig.java

```diff
@@ -218,6 +219,21 @@ private void checkStringToIntWithBoundaries(
     }
   }
 
+  // version validation and migration logic goes here
+  // e.g. unsupported versions would fail or be migrated
+  // in this release we support only the current app version
+  private void validateTabulatorVersion() {
+    if (isNullOrBlank(getTabulatorVersion())) {
+      isValid = false;
+      Logger.log(Level.SEVERE, "Tabulator version is required!");
+    } else {
+      if (!getTabulatorVersion().equals(Main.APP_VERSION)) {
+        isValid = false;
+        Logger.log(Level.SEVERE, "Tabulator version %s not supported!", getTabulatorVersion());
```

Comment:
Can we offer some recourse or advice here? CLI users who manually build their config files are kind of screwed...

Reply:
Should we say that the config file must set tabulatorVersion to %d?

Reply:
yeah sounds good.